pkg/operation/botanist/logging.go

```diff
@@ -20,156 +20,34 @@ import (
-
-		sgCuratorPassword     = b.Secrets["curator-sg-credentials"].Data[secrets.DataKeyPassword]
-		sgCuratorPasswordHash = b.Secrets["curator-sg-credentials"].Data[secrets.DataKeyPasswordBcryptHash]
-
-		sgUserPasswordHash  = b.Secrets[kibanaUserIngressCredentialsSecretName].Data[secrets.DataKeyPasswordBcryptHash]
-		sgAdminPasswordHash = b.Secrets[common.KibanaAdminIngressCredentialsSecretName].Data[secrets.DataKeyPasswordBcryptHash]
-
-		userIngressBasicAuth  = utils.CreateSHA1Secret(b.Secrets[kibanaUserIngressCredentialsSecretName].Data[secrets.DataKeyUserName], b.Secrets[kibanaUserIngressCredentialsSecretName].Data[secrets.DataKeyPassword])
-		adminIngressBasicAuth = utils.CreateSHA1Secret(b.Secrets[common.KibanaAdminIngressCredentialsSecretName].Data[secrets.DataKeyUserName], b.Secrets[common.KibanaAdminIngressCredentialsSecretName].Data[secrets.DataKeyPassword])
-		ingressBasicAuth      string
-
-		sgFluentdPasswordHash string
-	)
-
-	userIngressBasicAuthDecoded, err := utils.DecodeBase64(userIngressBasicAuth)
+	err := common.DeleteOldLoggingStack(ctx, b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 	if err != nil {
-		return err
+		return nil
 	}
 
-	adminIngressBasicAuthDecoded, err := utils.DecodeBase64(adminIngressBasicAuth)
-	if err != nil {
-		return err
+	if b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting || !gardenletfeatures.FeatureGate.Enabled(features.Logging) {
+		return common.DeleteLoggingStack(ctx, b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 	}
 
-	ingressBasicAuthDecoded := fmt.Sprintf("%s\n%s", string(userIngressBasicAuthDecoded), string(adminIngressBasicAuthDecoded))
-	ingressBasicAuth = utils.EncodeBase64([]byte(ingressBasicAuthDecoded))
-
 	images, err := b.InjectSeedSeedImages(map[string]interface{}{},
-		common.ElasticsearchImageName,
-		common.ElasticsearchMetricsExporterImageName,
-		common.ElasticsearchSearchguardImageName,
-		common.CuratorImageName,
-		common.KibanaImageName,
-		common.SearchguardImageName,
-		common.AlpineImageName,
+		common.LokiImageName,
 	)
 	if err != nil {
-		return err
-	}
-
-	ct := b.Shoot.Info.CreationTimestamp.Time
-
-	sgFluentdSecret := &corev1.Secret{}
-	if err = b.K8sSeedClient.Client().Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, "fluentd-es-sg-credentials"), sgFluentdSecret); err != nil {
-		return err
-	}
-
-	sgFluentdPasswordHash = string(sgFluentdSecret.Data["bcryptPasswordHash"])
-
-	kibanaTLSOverride := common.KibanaTLS
-	if b.ControlPlaneWildcardCert != nil {
-		kibanaTLSOverride = b.ControlPlaneWildcardCert.GetName()
-	}
-
-	hosts := []map[string]interface{}{
-		// TODO: timuthy - remove in the future. Old Kibana host is retained for migration reasons.
-		{
-			"hostName":   b.ComputeKibanaHostDeprecated(),
-			"secretName": common.KibanaTLS,
-		},
-		{
-			"hostName":   b.ComputeKibanaHost(),
-			"secretName": kibanaTLSOverride,
-		},
+		return nil
```

Comment:
Same as above: why not return the error? If there is a valid reason, would you mind adding a short comment to explain? Otherwise somebody might feel compelled to fix this in the future.

Reply:
fixed